evalml/utils/gen_utils.py

```diff
@@ -208,17 +208,23 @@ def _rename_column_names_to_numeric(X):
     Returns:
         Transformed X where column names are renamed to numerical values
     """
-    X_t = X
     if isinstance(X, (np.ndarray, list)):
         return pd.DataFrame(X)
+
     if isinstance(X, ww.DataTable):
         X_t = X.to_dataframe()
-        logical_types = X.logical_types
-    name_to_col_num = dict((col, col_num) for col_num, col in enumerate(list(X.columns)))
-    X_renamed = X_t.rename(columns=name_to_col_num, inplace=False)
+    else:
+        X_t = X.copy()
+
+    if len(X_t.columns) > 0 and isinstance(X_t.columns[0], tuple):
```

Comment:
Handling tuples. Working with WW directly doesn't allow renames unless passed another tuple, and this helps convert tuples to strings to 'flatten' the tuple. Ex: ('a', 1) --> '('a', 1)' to convert tuple multi-index cols to string cols

Reply:
I think the implementation here looks good! I would vote for `isinstance(X_t.columns, pd.MultiIndex)` to make it clearer what's we're guarding against. 

Looks like specifying tuples for column names automatically creates a MultiIndex:

![image](https://user-images.githubusercontent.com/41651716/106795925-b2d59980-6628-11eb-91fb-4412e1efc665.png)

Reply:
Yupperino, agreed that it'll make it more clear. Will update 😁